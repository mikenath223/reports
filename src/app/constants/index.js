export const pieColors = ["#A259FF", "#F24E1E", "#FFC107", "#6497B1"];
