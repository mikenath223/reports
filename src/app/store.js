import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger";
import projectsReducer from "./slices/projectsSlice";
import formReducer from "./slices/formSlice";
import userReducer from "./slices/userSlice";
import gatewayReducer from "./slices/gatewaySlice";
import reportsReducer from "./slices/reportsSlice";

const middlewares = [];
if (process.env.NODE_ENV === `development`) {
  middlewares.push(logger);
}

export const store = configureStore({
  reducer: {
    projects: projectsReducer,
    form: formReducer,
    users: userReducer,
    gateways: gatewayReducer,
    reports: reportsReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(middlewares),
});
