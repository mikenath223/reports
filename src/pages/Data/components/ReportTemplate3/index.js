import { formatDate } from "app/utils/formatDate";
import PropTypes from "prop-types";

export default function ReportTemplate3({ title, reports }) {
  const grandTotal = reports.reduce((a, c) => a + c.total, 0);

  return (
    <section className="reports-wrap">
      <div className="reports-template">
        <h1>{title}</h1>
        <div>
          <table className="table">
            <thead>
              <tr>
                <th className="date">Date</th>
                <th>Transaction ID</th>
                <th className="amount">Amount</th>
              </tr>
            </thead>
            <tbody>
              {reports?.map((project) => {
                return (
                  <tr key={project.amount}>
                    <td className="date">{formatDate(project?.created)}</td>
                    <td>{project?.paymentId}</td>
                    <td className="amount">{project?.amount} USD</td>
                  </tr>
                );
              })}
            </tbody>
          </table>
        </div>
      </div>
      <div className="grand-total">
        <span>
          TOTAL: {new Intl.NumberFormat("ja-JP").format(Math.round(grandTotal))}
        </span>
      </div>
    </section>
  );
}

ReportTemplate3.propTypes = {
  title: PropTypes.string,
  reports: PropTypes.arrayOf(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number])
  ),
};
