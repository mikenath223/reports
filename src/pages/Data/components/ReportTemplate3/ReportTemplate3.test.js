import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "app/store";
import ReportTemplate3 from ".";

describe("ReportTemplate3", function () {
  it("should match snapshot", () => {
    const props = {
      title: "Project 1",
      reports: [],
    };
    const { container } = render(
      <Provider store={store}>
        <ReportTemplate3 {...props} />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
