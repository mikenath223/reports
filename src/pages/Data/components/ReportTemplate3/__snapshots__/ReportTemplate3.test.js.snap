// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportTemplate3 should match snapshot 1`] = `
<div>
  <section
    class="reports-wrap"
  >
    <div
      class="reports-template"
    >
      <h1>
        Project 1
      </h1>
      <div>
        <table
          class="table"
        >
          <thead>
            <tr>
              <th
                class="date"
              >
                Date
              </th>
              <th>
                Transaction ID
              </th>
              <th
                class="amount"
              >
                Amount
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
    <div
      class="grand-total"
    >
      <span>
        TOTAL: 
        0
      </span>
    </div>
  </section>
</div>
`;
